src/app/shared/subscriptions/subscriptions-data.service.ts

```diff
@@ -0,0 +1,173 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, filter, map, switchMap, take } from 'rxjs/operators';
+
+
+import { NotificationsService } from '../notifications/notifications.service';
+import { RemoteDataBuildService } from '../../core/cache/builders/remote-data-build.service';
+import { RequestParam } from '../../core/cache/models/request-param.model';
+import { ObjectCacheService } from '../../core/cache/object-cache.service';
+import { DSOChangeAnalyzer } from '../../core/data/dso-change-analyzer.service';
+import { PaginatedList } from '../../core/data/paginated-list.model';
+import { RemoteData } from '../../core/data/remote-data';
+import { CreateRequest, PutRequest } from '../../core/data/request.models';
+import { FindListOptions } from '../../core/data/find-list-options.model';
+import { RestRequest } from '../../core/data/rest-request.model';
+
+import { RequestService } from '../../core/data/request.service';
+import { HALEndpointService } from '../../core/shared/hal-endpoint.service';
+import { Subscription } from './models/subscription.model';
+import { dataService } from '../../core/data/base/data-service.decorator';
+import { SUBSCRIPTION } from './models/subscription.resource-type';
+import { DSONameService } from '../../core/breadcrumbs/dso-name.service';
+import { NoContent } from '../../core/shared/NoContent.model';
+import { isNotEmpty, isNotEmptyOperator } from '../empty.util';
+
+import { getFirstCompletedRemoteData } from '../../core/shared/operators';
+import { sendRequest } from 'src/app/core/shared/request.operators';
+import { IdentifiableDataService } from '../../core/data/base/identifiable-data.service';
+import { DeleteDataImpl } from '../../core/data/base/delete-data';
+import { SearchDataImpl } from '../../core/data/base/search-data';
+import { FindAllData } from '../../core/data/base/find-all-data';
+import { followLink } from '../utils/follow-link-config.model';
+
+/**
+ * Provides methods to retrieve subscription resources from the REST API related CRUD actions.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+@dataService(SUBSCRIPTION)
+export class SubscriptionsDataService extends IdentifiableDataService<Subscription> {
```

Comment:
It looks like this service is missing specs.  I'm not seeing a `subscriptions-data.service.spec.ts` file.